### PR TITLE
Add horizontal rule

### PR DESCRIPTION
**Pull request by arronei**
Adding the functionality for markdown horizontal rules.

Fixes: #538

### CODE REVIEW

File: bikeshed/markdown.py [hunk 1]
```diff
@@ -67,6 +67,8 @@ def tokenizeLines(lines, numSpacesForIndentation, features=None, opaqueElements=
 			match = re.search(r"\{#([^ }]+)\}\s*$", line)
 			if match:
 				token['id'] = match.group(1)
+		elif re.match(r"([*_-])\s*\1\s*\1", line):
```
Comment (tabatkins) [inline bikeshed/markdown.py]: Need a `(\s*\1)*$` at the end to ensure that the line doesn't have anything else on it, per CommonMark.

File: bikeshed/markdown.py [hunk 2]
```diff
@@ -217,6 +221,11 @@ def parseMultiLineHeading(stream):
 	stream.advance(2)
 	return lines
 
+def parseHorizontalRule(stream):
+	lines =  ["<hr />\n"]
```
Comment (tabatkins) [inline bikeshed/markdown.py]: Just `<hr>`, please.  (I parse it as HTML anyway.)

**Comment by tabatkins** [comment]
Gave some nits; fix and I'll merge.

And lolwtf how did I get that crazy indentation.  Goddammit, Python, THIS IS WHY YOU SHOULD HAVE STANDARDIZED ON TABS.

**Comment by arronei** [comment]
I'm not seeing this error locally. The regex is valid from what I can tell. Any ideas what I'm missing?

**Comment by tabatkins** [comment]
Yes, the regex is valid. But it didn't fully match the CommonMark spec.

**Comment by arronei** [comment]
Got it. I'll change back to what I have with the tweak you noted.

**Comment by tabatkins** [comment]
All right, I tweaked the regex to make it not fail (which is weird, it should have worked), and fixed one line to match CommonMark more closely. All merged, thanks!

**Pull request merged**